neurons/validators/utils/uids.py

```diff
@@ -28,6 +28,36 @@ def check_uid_availability(
     # Available otherwise.
     return True
 
+def get_top_miner_uid(
+    metagraph: "bt.metagraph.Metagraph", vpermit_tao_limit: int = 4096, exclude: List[int] = None
+) -> torch.LongTensor:
+    """Returns the best available miner UID from the metagraph.
+    Args:
+        metagraph (bt.metagraph.Metagraph): Metagraph object
+        vpermit_tao_limit (int): Validator permit tao limit
+        exclude (List[int]): List of uids to exclude from the random sampling.
+    Returns:
+        best_miner_uid (torch.LongTensor): The best miner UID.    
+    """
+    candidate_uids = []
+    for uid in range(metagraph.n.item()):
```

Comment:
lets pick random miner from best x percentage,
that percentage should be configurable
we need to avoid to always pick the best miner (as it will be getting more and more positive score)
it will be up to validator what percentage of miners should be querable

Reply:
and what in situation, when given miner is picked up to be queried, but that miner has blacklisted our validator ?

Reply:
> lets pick random miner from best x percentage, that percentage should be configurable we need to avoid to always pick the best miner (as it will be getting more and more positive score) it will be up to validator what percentage of miners should be querable

thats cool, totally agree

Reply:
when select only one miner, if that doesn't response, there will be no response to show,
from this insight, i implemented the functionality to query top -n-miners and get responses, select one response randomly among received responses.

Reply:
> and what in situation, when given miner is picked up to be queried, but that miner has blacklisted our validator ?

in that case, cant receive any response

Reply:
ok, then maybe validator should also record for x amount of time that given miner should not be queried ?

Reply:
variable excluded_uids includes all miner UIDs which don't response. i set it as a global variable for mvp version. i think its enough for mvp. after test, i'll implement using redis.